deploy-cloud-run/action.yml

```diff
@@ -45,11 +63,134 @@ runs:
+      uses: fjogeleit/http-request-action@v1
+      if: ${{ inputs.blue_green_enabled == 'false' && inputs.test_url != '' }}
+      id: test_live
+      with:
+        url: ${{ inputs.test_url }}
+        method: 'GET'
+        ignoreSsl: true
+        ignoreStatusCodes: ${{ steps.http_responses.outputs.codes }}
+    - name: Logging - Deploy to Cloud Run Blue
+      run: echo "Deploying ${{ inputs.image_internal || inputs.image }} to ${{ inputs.service }} blue..."
+      if: ${{ inputs.blue_green_enabled == 'true' }}
+      shell: bash
+    - name: Deploy to Cloud Run Blue
+      uses: google-github-actions/deploy-cloudrun@v2
+      id: deploy_blue
+      if: ${{ inputs.blue_green_enabled == 'true' }}
+      with:
+        service: ${{ inputs.service }}
+        job: ${{ inputs.job }}
+        image: ${{ inputs.image_internal || inputs.image }}
+        region: ${{ inputs.region }}
+        tag: blue-release
+        no_traffic: true
+        skip_default_labels: true
+    - name: Run HTTP health check on blue
+      uses: fjogeleit/http-request-action@v1
+      if: ${{ inputs.blue_green_enabled == 'true' && inputs.test_url != ''}}
+      id: test_blue
+      with:
+        url: ${{ steps.deploy_blue.outputs.url }}${{ steps.parse_url.outputs.url_path }}
+        method: 'GET'
+        ignoreSsl: true
+        ignoreStatusCodes: ${{ steps.http_responses.outputs.codes }}
+    - name: Logging - Check HTTP Status from test_blue
+      run: |
+        echo "HTTP Status from test_blue: ${{ steps.test_blue.outputs.status }}"
+      if: ${{ inputs.blue_green_enabled == 'true' }}
+      shell: bash
+    - name: Logging - Deploy to Cloud Run Green
+      run: echo "Deploying ${{ inputs.image_internal || inputs.image }} to ${{ inputs.service }} green..."
+      if: ${{ inputs.blue_green_enabled == 'true' && steps.test_blue.outputs.status >= 200 && steps.test_blue.outputs.status < 300 }}
+      shell: bash
+    - name: Deploy to Cloud Run Green
+      uses: google-github-actions/deploy-cloudrun@v2
+      id: deploy_green
+      if: ${{ inputs.blue_green_enabled == 'true' && steps.test_blue.outputs.status >= 200 && steps.test_blue.outputs.status < 300 }}
+      with:
+        service: ${{ inputs.service }}
+        job: ${{ inputs.job }}
+        image: ${{ inputs.image_internal || inputs.image }}
+        region: ${{ inputs.region }}
+        revision_traffic: 'LATEST=100'
+        skip_default_labels: 'true'
+        update_traffic_flags: '--remove-tags blue-release'
+    - name: Run HTTP health check on green
+      continue-on-error: true
+      uses: fjogeleit/http-request-action@v1
+      if: ${{ inputs.blue_green_enabled == 'true' && inputs.test_url != '' && steps.test_blue.outputs.status >= 200 && steps.test_blue.outputs.status < 300 }}
+      id: test_green
+      with:
+        url: ${{ inputs.test_url }}
+        method: 'GET'
+        ignoreSsl: true
+        ignoreStatusCodes: ${{ steps.http_responses.outputs.codes }}
```

Comment:
@rot1024 for green/live testing it simulates external URL calling

Later I might also add functionality to supply additional parameters such as dynamic method, body, etc. But for now I think using simple GET to specific endpoint that determine the service liveness is sufficient